add ambient beta blog (#15070)
istio/istio.io: content/en/blog/2024/ambient-reaches-beta/index.md

---
title: "Istio’s new sidecar-less ‘Ambient Mode’ reaches Beta in version 1.22"
description: The latest release from Istio brings service mesh Layer 4 & 7 features to production readiness without sidecars.

> publishdate: 2024-05-14

Reviewer: Looks like 1.22 is aiming for today, so I think this should be the 13th as well?
```suggestion
publishdate: 2024-05-13
```

that introduced a new data plane mode in Istio that did not require sidecars, offering a simplified operational experience of
Istio. After 20 months of hard work and collaboration within the Istio community, with contributions from Solo.io, Google,
Microsoft, Intel, Aviatrix, Huawei, IBM, Red Hat, and others, we are excited to announce that ambient mode has reached Beta in
version 1.22! The beta release of 1.22 indicates the features of ambient mode are now ready for production workloads
with appropriate precautions. This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
readiness without sidecars.

## Why ambient mode?

We listened to the feedback from Istio users and observed a growing demand for mesh capabilities for their applications but
found the resource overhead and operational complexity of sidecars hard to overcome. Challenges that Istio sidecar users
have shared with us include: how Istio can break applications after sidecars are added, the large consumption of resources by
sidecars, and the inconvenience of the requirement to restart application pods with every new proxy release.

> Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application

Reviewer 2: I think it reads better if there's a comma between "application" and "alleviating".

Author: updated

> Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application

Author: ```suggestion
Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application,
```

alleviating the previous barriers of complexity faced by users looking to implement service mesh.

> In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.
> 
> As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named 'ambient mode'
> as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it
> and required no restarting of applications by users. You can refer to the [Introducing Ambient Mesh blog](/blog/2022/introducing-ambient-mesh/)
> for more information on why we started ambient mode in Istio.

Author: ```suggestion
As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named 'ambient mode'
as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it
and required no restarting of applications by users.
In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.

Refer to the [Introducing Ambient Mesh blog](/blog/2022/introducing-ambient-mesh/)
for more information on why we started ambient mode in Istio.
```

## How does ambient mode make adoption easier?

> Istio’s ambient mode introduces lightweight, shared node proxies and optional Layer 7 (L7) proxies and removes the need for

Reviewer 2: ```suggestion
Istio’s ambient mode introduces lightweight, shared node proxies and optional Layer 7 (L7) proxies, which removes the need for
```

traditional sidecar proxies from the data plane. The core innovation behind ambient mode is that it slices the L4 and L7
processing into two distinct layers. This layered approach allows you to adopt Istio incrementally, enabling a smooth
transition from no mesh, to a secure overlay (L4), to optional full L7 processing — on a per-namespace basis, as needed across
your fleet.

Ambient mode works without any modification required to your existing Kubernetes deployments. Users can label a namespace to
add all of its workloads to Istio’s ambient mode, or opt-out certain deployments as needed. By utilizing ambient mode, users
bypass some of the previously restrictive elements of the sidecar model and instead can now expect server send-first protocols

> to work, see most of the reserved ports are now removed, and the ability for containers to bypass the sidecar – either

Reviewer 2: ```suggestion
to work, see most of the reserved ports are removed, and the ability for containers to bypass the sidecar – either
```

Microsoft, Intel, Aviatrix, Huawei, IBM, Red Hat, and others, we are excited to announce that ambient mode has reached Beta in
version 1.22! The beta release of 1.22 indicates the features of ambient mode are now ready for production workloads
with appropriate precautions. This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
readiness without sidecars.

## Why ambient mode?

We listened to the feedback from Istio users and observed a growing demand for mesh capabilities for their applications but
found the resource overhead and operational complexity of sidecars hard to overcome. Challenges that Istio sidecar users
have shared with us include: how Istio can break applications after sidecars are added, the large consumption of resources by
sidecars, and the inconvenience of the requirement to restart application pods with every new proxy release.
Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application,
alleviating the previous barriers of complexity faced by users looking to implement service mesh.

> As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named 'ambient mode'
> as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it
> and required no restarting of applications by users.
> In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.
> 
> Refer to the [Introducing Ambient Mesh blog](/blog/2022/introducing-ambient-mesh/)
> for more information on why we started ambient mode in Istio.

Author: ```suggestion
In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.

As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named 'ambient mode'
as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it
and required no restarting of applications by users. Refer to the [Introducing Ambient Mesh blog](/blog/2022/introducing-ambient-mesh/)
for more information on why we started ambient mode in Istio.
```

---

> title: "Istio’s new sidecar-less ‘Ambient Mode’ reaches Beta in version 1.22"

Author: Should i use double quote here? @dhawton

Author: Ok cool, thanks! Understood the timing.

Reviewer 3: finally at the PC and it's merged already.  I have thoughts 😄 

"Ambient Mode" shouldn't be capitalised, it's not a brand, it shouldn't be in quotes, but should be emphasised more than "sidecar-less" (which probably shouldn't have a hyphen)

I'll go back to my hill...

Microsoft, Intel, Aviatrix, Huawei, IBM, Red Hat, and others, we are excited to announce that ambient mode has reached Beta in
version 1.22! The beta release of 1.22 indicates the features of ambient mode are now ready for production workloads
with appropriate precautions. This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
readiness without sidecars.

## Why ambient mode?

We listened to the feedback from Istio users and observed a growing demand for mesh capabilities for their applications but
found the resource overhead and operational complexity of sidecars hard to overcome. Challenges that Istio sidecar users
have shared with us include: how Istio can break applications after sidecars are added, the large consumption of resources by
sidecars, and the inconvenience of the requirement to restart application pods with every new proxy release.

As a community, we listened and designed ambient mode to tackle these problems,
alleviating the previous barriers of complexity faced by users looking to implement service mesh. This new feature from Istio

> was named 'ambient mode' as it was designed to be transparent to your application, ensuring no additional configuration was

Author: Should i use double quote here? @dhawton

Reviewer: I can see it going either way, personally.

Author: Ok cool, thanks. Will submit a PR if others complain!

> Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application,
> alleviating the previous barriers of complexity faced by users looking to implement service mesh. This new feature from Istio
> was named 'ambient mode' as it was designed to be transparent to your application, ensuring no additional configuration was
> required to adopt it and required no restarting of applications by users. In ambient mode it is trivial to add or remove
> applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications
> in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to
> restart applications.

Author: ```suggestion
As a community, we listened and designed ambient mode to tackle these problems,
alleviating the previous barriers of complexity faced by users looking to implement service mesh. This new feature from Istio
was named 'ambient mode' as it was designed to be transparent to your application, ensuring no additional configuration was
required to adopt it and required no restarting of applications by users. In ambient mode it is trivial to add or remove
applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications
in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to
restart applications.
```

for more information on why we started ambient mode in Istio.

## How does ambient mode make adoption easier?

Istio’s ambient mode introduces lightweight, shared node proxies and optional Layer 7 (L7) proxies and removes the need for
traditional sidecar proxies from the data plane. The core innovation behind ambient mode is that it slices the L4 and L7
processing into two distinct layers. This layered approach allows you to adopt Istio incrementally, enabling a smooth
transition from no mesh, to a secure overlay (L4), to optional full L7 processing — on a per-namespace basis, as needed across
your fleet.

Ambient mode works without any modification required to your existing Kubernetes deployments. Users can label a namespace to
add all of its workloads to Istio’s ambient mode, or opt-out certain deployments as needed. By utilizing ambient mode, users
bypass some of the previously restrictive elements of the sidecar model and instead can now expect server send-first protocols
to work, see most of the reserved ports are now removed, and the ability for containers to bypass the sidecar – either
maliciously or not – is eliminated.

> 

Reviewer 4: ```suggestion
```

The lightweight shared L4 node proxy is called ztunnel (zero-trust tunnel). Ztunnel drastically reduces the overhead of
running a mesh by removing the need to potentially over provision memory and CPU within a cluster to handle expected loads. In
some use cases, the savings can exceed 90% or more, while still providing zero-trust security using mutual TLS with
cryptographic identity, simple L4 authorization policies, and telemetry.

The L7 proxies are called waypoints. Waypoints process L7 functions such as traffic routing, rich authorization policy
enforcement, and enterprise-grade resilience. Waypoints run outside of your application deployments and can scale independently
based on your needs, which could be for the entire namespace or for multiple services within the namespace. Compared with
sidecars, you don’t need one waypoint per application pod, and you can scale your waypoint effectively based on its scope,
thus saving significant amounts of CPU and memory in most cases.

The separation between the L4 secure overlay layer and L7 processing layer allows incremental adoption of the ambient mode data

> plane in contrast to the earlier binary “all-in” injection of sidecars. Users can start with the secure overlay layer which

Author: ```suggestion
plane in contrast to the earlier binary "all-in" injection of sidecars. Users can start with the secure overlay layer which
```

---
title: "Istio’s new sidecar-less ‘Ambient Mode’ reaches Beta in version 1.22"
description: The latest release from Istio brings service mesh Layer 4 & 7 features to production readiness without sidecars.
publishdate: 2024-05-14
attribution: "Lin Sun (Solo.io), for the Istio Steering and Technical Oversight Committees"
keywords: [Istio Day,Istio,conference,KubeCon,CloudNativeCon]
---

Istio ambient service mesh was [announced in September 2022](/blog/2022/introducing-ambient-mesh/) as an experimental branch
that introduced a new data plane mode in Istio that did not require sidecars, offering a simplified operational experience of
Istio. After 20 months of hard work and collaboration within the Istio community, with contributions from Solo.io, Google,
Microsoft, Intel, Aviatrix, Huawei, IBM, Red Hat, and others, we are excited to announce that ambient mode has reached Beta in

> version 1.22! The beta release of 1.22 indicates the features of ambient mode are now stable and ready for production workloads

Author: ```suggestion
version 1.22! The beta release of 1.22 indicates the features of ambient mode are now ready for production workloads
```

with appropriate cautions. This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
readiness without sidecars.

## Why ambient mode?

> We listened to the feedback from Istio users and observed a growing demand for mesh capabilities for their applications but who found the resource overhead and operational complexity of sidecars hard to overcome.
> functions for their applications and found the complexity and overhead of sidecars from service meshes hard to overcome.

Author: ```suggestion
We listened to the feedback from Istio users and observed a growing demand for mesh capabilities for their applications but found the resource overhead and operational
complexity of sidecars hard to overcome.
```

Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application

> alleviating the previous barriers of complexity faced by users looking to implement service mesh.

Author: ```suggestion
alleviating the previous barriers of complexity faced by users looking to implement service mesh.
```

In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.

> your namespace with `istio.io/dataplane-mode=ambient`, and your application pods in the namespace are added to the mesh and the
> communications among your application pods in ambient are secured with mTLS, without restartarting or the need for sidecars!

Author: ```suggestion
```

---
title: "Istio’s new sidecar-less ‘Ambient Mode’ reaches Beta in version 1.22"
description: The latest release from Istio brings service mesh Layer 4 & 7 features to production readiness without sidecars.
publishdate: 2024-05-14
attribution: "Lin Sun (Solo.io), for the Istio Steering and Technical Oversight Committees"
keywords: [Istio Day,Istio,conference,KubeCon,CloudNativeCon]
---

Istio ambient service mesh was [announced in September 2022](/blog/2022/introducing-ambient-mesh/) as an experimental branch
that introduced a new data plane mode in Istio that did not require sidecars, offering a simplified operational experience of

> Istio. After 20 months of hard work and collaboration within the Istio community, with contributions from  Solo.io, Google,

Reviewer: ```suggestion
Istio. After 20 months of hard work and collaboration within the Istio community, with contributions from Solo.io, Google,
```

Microsoft, Intel, Aviatrix, Huawei, IBM, Red Hat, and others, we are excited to announce that ambient mode has reached Beta in

> version 1.22! The beta release of 1.22 indicates the features of ambient mode are now stable and ready for production workloads

Reviewer 5: +1 to @justinpettit's point. I feel like mixing beta and stable will confuse users

Author: I've removed "stable"... agreed it is confusing.

In istio's term, beta is when we start to recommend for productions with caution.  BTW, this wording was approved by steering before also: https://www.cncf.io/blog/2024/03/19/istio-announces-the-beta-release-of-ambient-mode/

Reviewer 5: This is much better; thanks!

> with appropriate cautions.  This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production

Reviewer 6: s/with appropraite xxxx/with some caveats.  (link to that whats in scope section below)

> with appropriate cautions.  This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production

Reviewer 2: I've not heard beta used to describe software as "stable and ready for production workloads".  In fact, the Istio documentation states that beta is ["Used to vet a solution in production without committing to it in the long term, to assess its viability, performance, usability, etc."](https://istio.io/latest/docs/releases/feature-stages/)  I realize that there is a caveat with "appropriate cautions", but I think that's understating the risk and what is usually being attempted with a beta release.

> with appropriate cautions.  This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production

Reviewer 4: ```suggestion
with appropriate cautions. This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
```

> with appropriate cautions.  This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production

Reviewer: ```suggestion
with appropriate precautions.  This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
```

Author: Feels it is cautions, will wait till others to weigh in.

Reviewer 4: `precautions` makes sense if we are referring to steps that a user will take to ensure we are carefully running production workloads.

`caution` a term meaning to avoid danger. In this case we are telling the user to be cautious when running ambient so I don't think `cautions` makes sense but to act with `caution` does. Wdyt?
```suggestion
with caution.  This is a huge milestone for Istio, bringing both Layer 4 and Layer 7 mesh features to production
```

Reviewer: The way I read it was steps to be taken before doing, which is a precaution. IE, in this case, we're still saying there are some elements that are alpha, which they should check beforehand.

Reviewer 4: Yeah that makes sense just wanted to point out that I don't think `cautions` is right if we went that route. But yeah, `precautions` seems to be the word we're looking for here.

Author: Thanks! updated to precautions!

readiness without sidecars.

## Why ambient mode?

> We listened to the feedback from the community and Istio users and observed a growing need for users wanting to implement mesh

Reviewer 6: ```suggestion
We listened to the feedback from Istio users and observed a growing demand for mesh capabilities for their applications but who found the resource overhead and operational complexity of sidecars hard to overcome.
```

> functions for their applications and found the complexity and overhead of sidecars from service meshes hard to overcome.

Reviewer 6: I think you need to collapse this setence into above, my comment applied to both. Alse shouldn't be a newline here.

> functions for their applications and found the complexity and overhead of sidecars from service meshes hard to overcome.

Reviewer 4: left over line/refactor? Seems to repeat a lot of what is at the end of the first sentence above it.

Istio’s ambient mode has been designed for simplified operations without requiring changes or restarts to your application

> alleviating the previous barriers of complexity faced by users looking to implement service mesh.

Reviewer 6: alleviating the previous barriers of complexity faced by users looking to implement service mesh.

Author: Sorry - seems same as what the texts have?

Author: think someone might have suggested a change and i accepted it earlier.

> Ambient mode also enables you to effortlessly enroll your applications onto Istio without sidecars. You can now simply label

Reviewer 6: ```suggestion
In ambient mode it is not trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.
```

Author: You mean it is trivial :) I'll update it.

> your namespace with `istio.io/dataplane-mode=ambient`, your applications in the namespace are added to the mesh and the

Reviewer: ```suggestion
your namespace with `istio.io/dataplane-mode=ambient`, and your application pods in the namespace are added to the mesh and the
```

> In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.

Reviewer 4: ```suggestion
In ambient mode it is trivial to add or remove applications from the mesh. You can now simply label your namespace with `istio.io/dataplane-mode=ambient` and all applications in the namespace are added to the mesh. This immediately secures all traffic with mTLS, all without sidecars or the need to restart applications.
```

communications among your application pods in ambient are secured with mTLS, without restartarting or the need for sidecars!

> Some challenges that Istio sidecar users have shared with us include; how Istio can break applications after sidecars are

Reviewer: ```suggestion
Some challenges that Istio sidecar users have shared with us include: how Istio can break applications after sidecars are
```

Reviewer 6: I don't think we want to lean in to a message that says 'break applications' too hard. Yes, it can happen, but its actually quite rare. Could instead talk about how we've made some 'transport improvements' that allow to fully 'support all types of TCP' traffic but even that is probably too much. 

You talk above about application restarts, lean into the 'pain' of the there and avoid repeating here?

> your namespace with `istio.io/dataplane-mode=ambient`, and your application pods in the namespace are added to the mesh and the

Reviewer 6: quoting of the label is rendering incorrectly

Author: should be fixed now.

> your namespace with `istio.io/dataplane-mode=ambient`, and your application pods in the namespace are added to the mesh and the

Reviewer 4: repeated line; matches the sentence before this

> added, or the large consumption of resources by sidecars, and the inconvenience of the requirement to restart application pods

Reviewer: ```suggestion
added, the large consumption of resources by sidecars, and the inconvenience of the requirement to restart application pods
```

> with every new proxy version.

Reviewer 6: "with every new proxy release."

Author: updated

> Some challenges that Istio sidecar users have shared with us include: how Istio can break applications after sidecars are

Reviewer 6: This paragrpah is pretty repetitive with the first one in this heading. 'listened to users' ... 'list of known pains'

Consider merging but retain the more explicit breakdown of the common pains

Author: Tried to merge it with the first paragraph in this section, let me know if you have other suggestions

As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named ambient

> mode as it was designed to be transparent to your application ensuring no additional configuration was required to adopt it

Reviewer: ```suggestion
mode as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it
```

> with every new proxy version.

Author: ```suggestion
with every new proxy release.
```

> As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named ambient

Reviewer 6: was named 'ambient mode' ...

> As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named ambient
> mode as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it

Author: ```suggestion
As a community, we listened and designed ambient mode to tackle these problems. This new feature from Istio was named 'ambient mode'
as it was designed to be transparent to your application, ensuring no additional configuration was required to adopt it
```

> and required no restarting of applications by users. You can refer to the [introducing ambient mesh blog](/blog/2022/introducing-ambient-mesh/)

Reviewer 6: Title Case for the blog reference "Introducing Ambient Mesh"

> and required no restarting of applications by users. You can refer to the [introducing ambient mesh blog](/blog/2022/introducing-ambient-mesh/)

Author: ```suggestion
and required no restarting of applications by users. You can refer to the [Introducing Ambient Mesh blog](/blog/2022/introducing-ambient-mesh/)
```

for more information on why we started ambient mode in Istio.

## How does ambient mode make adoption easier?

Istio’s ambient mode introduces lightweight, shared node proxies and optional Layer 7 (L7) proxies and removes the need for
traditional sidecar proxies from the data plane. The core innovation behind ambient mode is that it slices the L4 and L7
processing into two distinct layers. This layered approach allows you to adopt Istio incrementally, enabling a smooth
transition from no mesh, to a secure overlay (L4), to optional full L7 processing — on a per-namespace basis, as needed across

> workloads.

Reviewer 6: s/workloads/your fleet

Author: updated!

> Ambient mode works without any modification required from your existing Kubernetes deployments. Users can label a namespace to

Reviewer: ```suggestion
Ambient mode works without any modification required to your existing Kubernetes deployments. Users can label a namespace to
```

> transition from no mesh, to a secure overlay (L4), to optional full L7 processing — on a per-namespace basis, as needed across

Reviewer 6: nice!

Author: Thanks!

> add all of its workloads to Istio’s ambient mode, or opt-out of certain deployments as needed. By utilizing ambient mode, users

Reviewer 6: opt-out certain deployments as needed.   (remove the 'of')

Author: updated

> add all of its workloads to Istio’s ambient mode, or opt-out of certain deployments as needed. By utilizing ambient mode, users

Author: ```suggestion
add all of its workloads to Istio’s ambient mode, or opt-out certain deployments as needed. By utilizing ambient mode, users
```

bypass some of the previously restrictive elements of the sidecar model and instead can now expect server send-first protocols
to work, see most of the reserved ports are now removed, and the ability for containers to bypass the sidecar – either

> maliciously or not – is greatly reduced.

Reviewer 6: this lanuage is pretty equivocal. I'd say we just say 'eliminated' and be done with it.

Author: I think I had eliminated earlier, then @yuval-k pointed out there is possible way to bypass (for egress i think?) which the community has plans to mitigate the risk.  I can change back to 'eliminated' if we don't feel it needs to be exposed in the beta blog.

Author: updated to eliminated.  please reopen if you feel strongly @yuval-k

> maliciously or not – is greatly reduced.

Author: ```suggestion
maliciously or not – is eliminated.
```

> As part of the 1.22 release of Istio, the announcement of ambient mode in Beta also includes how this latest release brings

Reviewer 6: I think you can remove this whole paragraph, its just repeating something that has been said a few times already

Author: done

> both Layer 4 and Layer 7 mesh features to production readiness without sidecars.

Reviewer 2: As stated earlier, I don't think beta indicates "production readiness".

Author: This paragraph is removed.

> The lightweight shared L4 node proxy is called the zero-trust tunnel, or ztunnel. Ztunnel drastically reduces the overhead of

Author: ```suggestion
The lightweight shared L4 node proxy is called ztunnel (zero-trust tunnel). Ztunnel drastically reduces the overhead of
```

> The lightweight shared L4 node proxy is called the zero-trust tunnel, or ztunnel. Ztunnel drastically reduces the overhead of

Reviewer 6: is called ztunnel (zero-trust tunnel)..

Author: updated

running a mesh by removing the need to potentially over provision memory and CPU within a cluster to handle expected loads. In

> some use cases, the savings can exceed 90% or more, while still providing zero-trust security using mutual TLS with

Reviewer 6: is there anything public we can link to that shows the saving?

@howardjohn @craigbox

Author: we have a blog from Solo we could show, @ably77 is also working on https://github.com/istio/istio.io/issues/14957.

Reviewer: Leaving this as-is for now. Can follow up once that blog is posted with a link to it.

cryptographic identity, simple L4 authorization policies, and telemetry.

The L7 proxies are called waypoints. Waypoints process L7 functions such as traffic routing, rich authorization policy
enforcement, and enterprise-grade resilience. Waypoints run outside of your application deployments and can scale independently
based on your needs, which could be for the entire namespace or for multiple services within the namespace. Compared with

> sidecars, you don’t need 1 waypoint proxy per application pod, and you can scale your waypoint effectively based on its scope,

Reviewer: ```suggestion
sidecars, you don’t need one waypoint per application pod, and you can scale your waypoint effectively based on its scope,
```

Reviewer 6: ... scale your waypoints based on the amount of traffic they handle.

> thus saving CPU and memory significantly in many cases.

Reviewer 6: thus saving significant amounts of CPU and memory in most cases.

The separation between the L4 secure overlay layer and L7 processing layer allows incremental adoption of the ambient mode data

> plane in contrast to the earlier binary “all in” injection of sidecars. Users can start with the secure overlay layer which

Reviewer: ```suggestion
plane in contrast to the earlier binary “all-in” injection of sidecars. Users can start with the secure overlay layer which
```

offers mTLS with cryptographic identity, simple L4 authorization policy, and telemetry. Later on, complex L7 handling such as
retries, traffic splitting, complex load balancing, and observability collection can be enabled on a case-by-case basis.

## What is in the scope of the Beta?

> We recommend you explore the following Beta functions of ambient mode in production with appropriate caution, after validating

Reviewer: ```suggestion
We recommend you explore the following Beta functions of ambient mode in production with appropriate precautions, after validating
```

them in test environments:

- [Install](/docs/ambient/install/).
- [Adding your workloads to the mesh](/docs/ambient/usage/add-workloads/) to gain mutual TLS with cryptographic identity, [L4 authorization policies](t/docs/ambient/usage/l4-policy/), and telemetry.

> - [Configure waypoint proxies](/docs/ambient/usage/waypoint/) to [use L7 functions](/docs/ambient/usage/l7-features/) such as traffic shifting, request routing, rich authorization policy enforcement.

Reviewer: ```suggestion
- [Configure waypoints](/docs/ambient/usage/waypoint/) to [use L7 functions](/docs/ambient/usage/l7-features/) such as traffic shifting, request routing, rich authorization policy enforcement.
```

Nit for consistency, these are called waypoints throughout the blog but then `waypoint proxy` here. We should either call them `waypoint proxy` or `waypoint`.

- Istio ingress gateway can work with workloads in ambient mesh supporting all existing Istio APIs.
- Use `istioctl` to operate waypoints, and troubleshoot ztunnel & waypoints.

### Alpha features

Other features we want to include in ambient mode have been implemented but remain in Alpha status in this release. Please help
test them, so they can be promoted to Beta in 1.23 or later:

- Multi-cluster installations
- DNS proxying
- Interoperability with sidecars
- IPv6/Dual stack
- SOCKS5 support (for outbound)
- Istio’s classic APIs (`VirtualService` and `DestinationRule`)

### Roadmap

We have a number of features which are not yet implemented in ambient mode but are planned for upcoming releases:

- Controlled egress traffic
- Multi-network support
- Improve `status` messages on resources to help troubleshoot and understand the mesh
- VM support

## What about sidecars?

Sidecars are not going away, and remain first-class citizens in Istio. You can continue to use sidecars and they will remain
fully supported.  For any feature outside of the Alpha or Beta scope for ambient mode, you should consider using the sidecar

> pattern before the feature is added to ambient mode. Some use cases, such as traffic shifting based on source labels, will

Author: ```suggestion
pattern until the feature is added to ambient mode. Some use cases, such as traffic shifting based on source labels, will
```

> pattern before the feature is added to ambient mode. Some use cases, such as traffic shifting based on source labels, will

Reviewer 2: The preliminary docs refer to "sidecar mode" as opposed to "sidecar pattern".  It might be good to use a consistent name to not confuse users.

I think "until" instead of "before".  (But that does imply that all sidecar features will be supported by ambient, which will likely not be true.)

Author: updated

> continue to be best implemented using the sidecar pattern. While we believe most use cases will be best served with a mesh in

Author: ```suggestion
continue to be best implemented using the sidecar mode. While we believe most use cases will be best served with a mesh in
```

ambient mode, the Istio project remains committed to ongoing sidecar mode support.

## Try Istio’s new sidecar-less ambient mode today!

With the 1.22 release of Istio and the beta release of ambient mode, it will be easier than ever to try out Istio on your own

> workloads. Follow the [get started guide](/docs/ambient/getting-started/) to explore ambient or [user guide](https://preliminary.istio.io/latest/docs/ambient/usage/)

Reviewer: ```suggestion
workloads. Follow the [getting started guide](/docs/ambient/getting-started/) to explore ambient or [user guide](https://preliminary.istio.io/latest/docs/ambient/usage/)
```

We'll need to update this link once the prelim docs are merged so that we link to the live site.

> workloads. Follow the [getting started guide](/docs/ambient/getting-started/) to explore ambient or [user guide](https://preliminary.istio.io/latest/docs/ambient/usage/)

Author: ```suggestion
workloads. Follow the [getting started guide](/docs/ambient/getting-started/) to explore ambient or [user guide](/docs/ambient/usage/)
```

> to learn how to incrementally adopt ambient for mutual TLS & L4 authorization policy, or traffic management, rich L7

Reviewer: ```suggestion
to learn how to incrementally adopt ambient for mutual TLS & L4 authorization policy, traffic management, rich L7
```

> authorization policy and more. Engage with us in the #ambient channel on our [Slack](slack.istio.io) or our discussion forum on

Reviewer: ```suggestion
authorization policy, and more. Engage with us in the #ambient channel on our [Slack](https://slack.istio.io) or our discussion forum on
```
